Use the official Guacamole Docker image

jsf9k opened the pull request:
## 🗣 Description ##

This pull request modifies the `docker-compose.yml` file to use [the official `guacamole/guacamole` Docker image](https://hub.docker.com/r/guacamole/guacamole) instead of [our custom `cisagov/guacamole` image](https://hub.docker.com/r/cisagov/guacamole).  We were using the custom Guacamole image because the official one lacked support for HTTP header authentication, but as of version 1.3.0 the official image includes the necessary support - added via [apache/guacamole#548](https://github.com/apache/guacamole-client/pull/548).

While I was mucking around in here, I went ahead and cleaned up a few things:
* Reduced the `version` in the `docker-compose.yml` file to the minimum required
* Deleted a duplicate `secrets` directory
* Added a `requirements-dev.txt` file to install `semver` for versioning

## 💭 Motivation and Context ##

We should use the official Docker image now that it is possible, if only to avoid the hassle of maintaining our own custom image.

## 🧪 Testing ##

I transferred this `docker-compose.yml` file to env0 in our staging COOL environment and verified that it worked as expected.

## ✅ Checklist ##

* [x] This PR has an informative and human-readable title.
* [x] Changes are limited to a single goal - _eschew scope creep!_
* [x] All relevant type-of-change labels have been added.
* [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
* [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
* [x] All relevant repo and/or project documentation has been updated to reflect the changes in this PR.
* [x] All new and existing tests pass.

jsf9k committed:
Use the official Guacamole Docker image

We were using a custom image, since the official one lacked support for HTTP header authentication.  This has been remedied as of version 1.3.0.

jsf9k committed:
Reduce docker-compose file version to minimum required

jsf9k committed:
Remove blurb about using a custom Guacamole Docker image

We are now able to use the official Guacamole Docker image.

jsf9k committed:
Remove duplicate secrets directory

jsf9k committed:
Add requirements-dev.txt

We require semver for development.

jsf9k committed:
Bumping version from 0.0.1 to 0.1.0

dav3r reviewed (approved):
Excellent- thanks for taking care of this! 🥑 

We should test if this new image fixes https://github.com/cisagov/cool-system/issues/123.

jsf9k commented:
> We should test if this new image fixes [cisagov/cool-system#123](https://github.com/cisagov/cool-system/issues/123).

Ah, yes.  I had forgotten about that.

FYI, I will build a new AMI once this PR is merged.

hillaryj reviewed (approved):
Great work! 🥇

mcdonnnj commented on docker-compose.yml:
```diff
@@ -1,5 +1,5 @@
 ---
-version: "3.7"
+version: "3.2"
```
Not to nitpick, but does this actually use any `3.2` features?

jsf9k replied:
I think you're right @mcdonnnj.  I was thinking that secrets was added in 3.2, but it was actually added in 3.1.  Do you concur that the version should be 3.1?

jsf9k replied:
See commit 75388d5.

mcdonnnj replied:
`3.1` is what I was thinking as well.

mcdonnnj reviewed (approved):
Thanks for tracking this and getting us updated! 💪 I had a minor question that doesn't block approval from me.

jsf9k committed:
Set docker-compose version to 3.1

@mcdonnnj pointed out that this docker-compose file does not actually
use any 3.2 features.  Secrets, which it does use, were added in
version 3.1; hence that is the minimum version required.

Co-authored-by: Nick M <50747025+mcdonnnj@users.noreply.github.com>

mcdonnnj reviewed (approved):
![image](https://user-images.githubusercontent.com/50747025/104341722-1c4e0680-54c8-11eb-8f42-b46f4efca1fb.png)
